sympy/physics/mechanics/kane.py

```diff
@@ -121,48 +160,60 @@ class KanesMethod(object):
     perform linearization and how to deal with dependent coordinates & speeds,
     and how do deal with bringing non-contributing forces into evidence.
 
+    References
+    ==========
+
+    .. [Kane1985] Kane, T., Levinson, D. Dynamics Theory and Applications. 1985
+       McGraw-Hill
+
     """
 
     def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
                  configuration_constraints=None, u_dependent=None,
                  velocity_constraints=None, acceleration_constraints=None,
                  u_auxiliary=None):
-        """
-        n : number of independent generalized coordinates
-        m : number of dependent generalized speeds and non-holonomic constraints
-        o : number of dependent generalized coordinates and holonomic constraints
-        p : number of independent generalized speeds
+        """Initializes a KanesMethod object.
 
         Parameters
         ==========
+
         frame : ReferenceFrame
             The inertial reference frame with which to form the equations of
             motion with respect to. All bodies and particles must have their
             velocity defined with respect to this frame, either directly or
             indirectly.
-        q_ind : ordered iterable of SymPy functions of time, len(n)
-            The n independent coordinates.
-        u_ind : ordered iterable, len(p)
+        q_ind : ordered iterable of functions of time, len(n)
+            The n independent generalized coordinates.
+        u_ind : ordered iterable of functions of time, len(p)
             The p independent generalized speeds.
         kd_eqs : iterable of SymPy expressions, optional
             The left hand side of the zero equal kinematical differential
-            equations. If None, the simple definition of u = q' is used.
-        q_dependent : ordered iterable, len(o)
-            The
-        configuration_constraints : iterable of SymPy expressions, optional
-        u_dependent : ordered iterable, len(m)
-            The dependent speeds.
-        velocity_constraints : iterable of SymPy expressions, optional
+            equations. (TODO: check if following statement is true) If None,
```

Comment:
The absence of kd_eqs is handled on line 435 (green side) where it simply initializes empty matrices and sets `self._qdot_u_map` to `None`. `self._qdot_u_map` is tested against `None` when determining if there are kinematic differential equations to return from the `kinddiffdict` (or something like that) but is also tested on lines 333 and 350 (also green side) and it is there that I believe you can determine the truth of the statement.